lib/net.cpp

```diff
@@ -15,6 +15,26 @@
 
 // https://wiki.openssl.org/index.php/SSL/TLS_Client
 
+static std::string socket_error(int err)
+{
+    static char msgbuf[256] = { '\0' };
+#ifdef _WIN32
+    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                    NULL,
+                    err,
+                    MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                    msgbuf,
+                    sizeof(msgbuf),
+                    NULL);
+#else
+    snprintf(msgbuf, sizeof(msgbuf), "%s", strerror(err));
+#endif
+    if (msgbuf[0] == '\0') {
```

Comment:
This seems like a redundant test, as I can see no case where the error message function succeeds but still returns an empty message. Instead, check for success of FormatMessage. (Note that strerror() has no error return value and is documented to always return a valid string.)

Reply:
Removed unnecessary check, and simplified the logic.